<?xml version="1.0"?>
<sdf version="1.9">
  <model name="ydlidar_g4">
    <pose>0 0 0 0 0 0</pose>

    <link name="lidar_link_1">

      <sensor name="g4_sensor" type="gpu_lidar">
        <update_rate>10</update_rate>
        <topic>lidar/scan</topic>

        <lidar>
          <scan>
            <horizontal>
              <samples>720</samples>
              <resolution>1</resolution>
              <min_angle>-3.14</min_angle>
              <max_angle>3.14</max_angle>
            </horizontal>
          </scan>

          <range>
            <min>0.1</min>
            <max>12.0</max>
            <resolution>0.01</resolution>
          </range>
        </lidar>

        <always_on>1</always_on>
        <visualize>true</visualize>
      </sensor>

    </link>

  </model>
</sdf>
